docs/operations/auth-ldap.md

```diff
@@ -22,182 +23,279 @@ title: "LDAP auth"
 ```
 
-To add additional roles to the group mapping, use the following API:
+The following example request adds the role `queryRole` to the `group1map` mapping:
 
 ```
-curl -i -v  -H "Content-Type: application/json" -u internal -X POST http://localhost:8081/druid-ext/basic-security/authorization/db/ldapauth/groupMappings/group1/roles/<newrole> 
+curl -i -v  -H "Content-Type: application/json" -u internal -X POST http://localhost:8081/druid-ext/basic-security/authorization/db/ldapauth/groupMappings/group1/roles/queryrole
 ```
 
-In the next two steps you will be creating a user, and assigning previously created roles to it. These steps are only needed in the following cases: 
-                                                                                                 
- - Your LDAP server does not support the `memberOf` attribute, or 
- - You want to configure a user with additional roles that are not mapped to the group(s) that the user is a member of
- 
- If this is not the case for your scenario, you can skip these steps.
+### Step 4: Add an LDAP user to Druid and assign a role
 
-### Step 4. Create a user
+You only need to complete this step if:
+- Your LDAP server doesn't support the `memberOf` attribute, or
+- You want to configure a user with additional Druid roles that are not mapped to the LDAP groups that the user belongs to.
 
-Once LDAP is enabled, only user passwords are verified with LDAP. You add the LDAP user to Druid as follows: 
+Example request to add the LDAP user `myuser` to Druid:
 
 ```
-curl -i -v  -H "Content-Type: application/json" -u internal -X POST http://localhost:8081/druid-ext/basic-security/authentication/db/ldap/users/<AD user> 
+curl -i -v  -H "Content-Type: application/json" -u internal -X POST http://localhost:8081/druid-ext/basic-security/authentication/db/ldap/users/myuser 
 ```
 
-### Step 5. Assign the role to the user 
-
-The following command shows how to assign a role to a user:
+Example request to assign the `myuser` user to the `queryRole` role:
 
 ```
-curl -i -v  -H "Content-Type: application/json" -u internal -X POST http://localhost:8081/druid-ext/basic-security/authorization/db/ldapauth/users/<AD user>/roles/<rolename> 
+curl -i -v  -H "Content-Type: application/json" -u internal -X POST http://localhost:8081/druid-ext/basic-security/authentication/db/ldap/users/myuser/roles/queryRole
 ```
 
-For more information about security and the basic security extension, see [Security Overview](security-overview.md). 
+## Enable LDAP over TLS (LDAPS)
+
+Once you've configured LDAP authentication in Druid, you can optionally make LDAP traffic confidential and secure by using Transport Layer Security (TLS)&mdash;previously Secure Socket Layer(SSL)&mdash;technology. 
+
+## Prerequisites
+
+Before you start to set up LDAPS in Druid:
+
+- You must [configure Druid for LDAP authentication](#configure-druid-for-ldap-authentication).
+- You need a CA certificate for your LDAP server.
```

Comment:
I get a little lost in the certificate section.

We are trying to establish trust between Druid and the LDAP server.

The LDAP server should be secured with a certificate issued by a public CA (certificate authority) or self-signed/certified by an internal CA (which is common case for internal-only services like LDAP)

You have to  import into the Druid trust store:
a) The LDAP server public certificate (regardless of who signed it)
b) the root certificate for the the CA that signed the certificate for the LDAP server. If it is a common public CA, this certificate may already be in the java truststore. Otherwise you need to import the certificate for the CA.

If you import the Root CA certificate, you're indicating you trust any certificate signed by that authority. When you import the LDAP server certificate, you're indicating that you trust the LDAP server based upon the authority of the CA

Reply:
Updated this section.